sys/riotboot/hdr.c

```diff
@@ -38,28 +39,106 @@
 
 #if __BYTE_ORDER__ != __ORDER_LITTLE_ENDIAN__
 #   error \
-    "This code is implementented in a way that it will only work for little-endian systems!"
+    "This code is implemented in a way that it will only work for little-endian systems!"
 #endif
 
+static uint32_t *_magic_number(riotboot_hdr_t *riotboot_hdr)
+{
+    return &riotboot_hdr->v1.magic_number;
+}
+
+static uint32_t *_version(riotboot_hdr_t *riotboot_hdr)
+{
+    switch (riotboot_hdr->v1.magic_number) {
+        case RIOTBOOT_MAGIC_V1:
+            return &riotboot_hdr->v1.version;
+        case RIOTBOOT_MAGIC_V2:
+            return &riotboot_hdr->v2.version;
+        default:
+            return NULL;
```

Comment:
```suggestion
    switch (riotboot_hdr->v1.magic_number) {
    case RIOTBOOT_MAGIC_V1:
        return &riotboot_hdr->v1.version;
    case RIOTBOOT_MAGIC_V2:
        return &riotboot_hdr->v2.version;
    default:
        return NULL;
```

As per our Coding Convention: https://github.com/RIOT-OS/RIOT/blob/master/CODING_CONVENTIONS.md#indentation-and-braces